Source/Foundation/NSObject.swift

```diff
@@ -48,7 +54,9 @@ extension NSObject {
     public func rex_classProperty<T: AnyObject>(keyPath: StaticString, placeholder: () -> T) -> MutableProperty<T> {
         return associatedProperty(self, keyPath: keyPath, placeholder: placeholder)
     }
+}
 
+extension NSObjectType where Self: NSObject, InstanceType == Self {
```

Comment:
Although given your improvements to `associatedProperty` I'm not sure that any of `rex_*Property` methods are worth keeping. They only save the `self` parameter and now require declaring `InstanceType` in each extension.